src/humancellatlas/data/metadata/helpers/staging_area.py

```diff
@@ -219,117 +207,103 @@ def _entity_ids_by_type(self,
 
 @attr.s(frozen=True, kw_only=True, auto_attribs=True)
 class GitHubStagingAreaFactory:
-    repo: Repository
-    ref: str
-    path: Tuple[str, ...]
+    url: str
+    repo: git.Repo
+    commit: git.Commit
+    path: tuple[str, ...]
```

Comment:
Perhaps

```suggestion
    path: Sequence[str]
```

would be more appropriate?

Reply:
Sames goes for any method parameters